### PR TITLE
Feature/feature icon medal

### PR DESCRIPTION
**Pull request by oricat8**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/entity/medalmanager/Medal.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243&bid=18472126)

File: src/main/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestController.java [hunk 1]
```diff
@@ -33,7 +34,7 @@
      */
     @PostMapping("/signup")
     @ResponseStatus(HttpStatus.CREATED)
-    public void signUp(@RequestBody Map<String, Object> user) throws DatabaseAccessException, FirebaseAuthException {
+    public void signUp(@RequestBody Map<String, Object> user) throws DatabaseAccessException, FirebaseAuthException, DocumentException {
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestController.java:37]: Codacy Issue found: [Line is longer than 120 characters (found 136).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/service/usermanager/UserServiceImpl.java [hunk 2]
```diff
@@ -29,7 +29,7 @@ public UserServiceImpl() {
     }
 
     @Override
-    public void createUser(String uid, UserEntity userEntity) throws DatabaseAccessException, FirebaseAuthException {
+    public void createUser(String uid, UserEntity userEntity) throws DatabaseAccessException, FirebaseAuthException, DocumentException {
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/service/usermanager/UserServiceImpl.java:32]: Codacy Issue found: [Line is longer than 120 characters (found 136).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java [hunk 3]
```diff
@@ -81,6 +87,7 @@ public void createUser(String uid, UserEntity userEntity) throws DatabaseAccessE
             batch.set(users.document(uid), userEntity);
             try {
                 batch.commit().get();
+                ((UserMedalDaoImpl)userMedalDao).createAllUserMedals(username);
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java:90]: Codacy Issue found: ['typecast' is not followed by whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java [hunk 4]
```diff
@@ -16,8 +16,11 @@
 import org.pesmypetcare.webservice.builders.Collections;
 import org.pesmypetcare.webservice.builders.Path;
 import org.pesmypetcare.webservice.dao.appmanager.StorageDao;
+import org.pesmypetcare.webservice.dao.medalmanager.UserMedalDao;
+import org.pesmypetcare.webservice.dao.medalmanager.UserMedalDaoImpl;
 import org.pesmypetcare.webservice.dao.petmanager.PetDao;
 import org.pesmypetcare.webservice.dao.petmanager.PetDaoImpl;
+import org.pesmypetcare.webservice.entity.medalmanager.UserMedalEntity;
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java:23]: Codacy Issue found: [Unused import - org.pesmypetcare.webservice.entity.medalmanager.UserMedalEntity.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDao.java [hunk 5]
```diff
@@ -20,7 +20,7 @@
      * @throws DatabaseAccessException If an error occurs when accessing the database
      * @throws FirebaseAuthException If an error occurs when retrieving the data
      */
-    void createUser(String uid, UserEntity userEntity) throws DatabaseAccessException, FirebaseAuthException;
+    void createUser(String uid, UserEntity userEntity) throws DatabaseAccessException, FirebaseAuthException, DocumentException;
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDao.java:23]: Codacy Issue found: [Line is longer than 120 characters (found 128).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/service/usermanager/UserService.java [hunk 6]
```diff
@@ -21,7 +21,7 @@
      * @throws DatabaseAccessException If an error occurs when accessing the database
      * @throws FirebaseAuthException If an error occurs when retrieving the data
      */
-    void createUser(String uid, UserEntity userEntity) throws DatabaseAccessException, FirebaseAuthException;
+    void createUser(String uid, UserEntity userEntity) throws DatabaseAccessException, FirebaseAuthException, DocumentException;
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/service/usermanager/UserService.java:24]: Codacy Issue found: [Line is longer than 120 characters (found 128).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/entity/medalmanager/Medal.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243&bid=18472126)

File: src/main/java/org/pesmypetcare/webservice/service/usermanager/UserService.java (hunk 6 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/service/usermanager/UserService.java:24]: Codacy Issue found: [Line is longer than 120 characters (found 128).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java (hunk 3 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java:90]: Codacy Issue found: ['typecast' is not followed by whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDao.java (hunk 5 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDao.java:23]: Codacy Issue found: [Line is longer than 120 characters (found 128).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java (hunk 4 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java:23]: Codacy Issue found: [Unused import - org.pesmypetcare.webservice.entity.medalmanager.UserMedalEntity.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestController.java [hunk 7]
```diff
@@ -4,6 +4,7 @@
 import org.pesmypetcare.webservice.entity.usermanager.User;
 import org.pesmypetcare.webservice.entity.usermanager.UserEntity;
 import org.pesmypetcare.webservice.error.DatabaseAccessException;
+import org.pesmypetcare.webservice.error.DocumentException;
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestController.java:7]: Codacy Issue found: [Unused import - org.pesmypetcare.webservice.error.DocumentException.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/service/usermanager/UserServiceImpl.java (hunk 2 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/service/usermanager/UserServiceImpl.java:32]: Codacy Issue found: [Line is longer than 120 characters (found 136).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/entity/medalmanager/Medal.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243&bid=18472126)

File: src/test/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoTest.java [hunk 8]
```diff
@@ -22,6 +22,8 @@
 import org.mockito.junit.jupiter.MockitoExtension;
 import org.pesmypetcare.webservice.builders.Collections;
 import org.pesmypetcare.webservice.dao.appmanager.StorageDao;
+import org.pesmypetcare.webservice.dao.medalmanager.UserMedalDao;
+import org.pesmypetcare.webservice.dao.medalmanager.UserMedalDaoImpl;
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoTest.java:26]: Codacy Issue found: [Unused import - org.pesmypetcare.webservice.dao.medalmanager.UserMedalDaoImpl.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java [hunk 9]
```diff
@@ -16,8 +16,11 @@
 import org.pesmypetcare.webservice.builders.Collections;
 import org.pesmypetcare.webservice.builders.Path;
 import org.pesmypetcare.webservice.dao.appmanager.StorageDao;
+import org.pesmypetcare.webservice.dao.medalmanager.UserMedalDao;
+import org.pesmypetcare.webservice.dao.medalmanager.UserMedalDaoImpl;
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java:20]: Codacy Issue found: [Unused import - org.pesmypetcare.webservice.dao.medalmanager.UserMedalDaoImpl.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/service/usermanager/UserService.java (hunk 6 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/service/usermanager/UserService.java:24]: Codacy Issue found: [Line is longer than 120 characters (found 128).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDao.java (hunk 5 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDao.java:23]: Codacy Issue found: [Line is longer than 120 characters (found 128).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java (hunk 4 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java:23]: Codacy Issue found: [Unused import - org.pesmypetcare.webservice.entity.medalmanager.UserMedalEntity.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/entity/medalmanager/Medal.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5641243&bid=18472126)

**Review by marclll — APPROVED** [review]

**Review by santidrj — APPROVED** [review]
:+1:

**Pull request merged**